src/storage_proxy.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Class to define the current storage instance to be used at the moment
+ *
+ * Any storage access will be done by StorageProxy.getStorage().method
+ * and this allow many storage instances to exist.
+ *
+ * Anytime you want to change the current storage instance to be used,
+ * just need to call StorageProxy.setStorage(storage)
+ *
+ * @class
+ * @name StorageProxy
+ */
+class StorageProxy {
```

Comment:
I don't understand why we had to create this. Couldn't we use `storage.js` for this? We'd use `storage.setStore()` instead of `storageProxy.setStorage`.

Reply:
The problem is that the `storage` is used all across the code. Before a major refactor to stop doing it, we need to be able to change the storage without while we keep the lib compatible to legacy code in our wallets.

Reply:
I still don't understand the problem.

Reply:
We have many storage instances, we need a place to store what storage should be used in the moment.

For example:

`w1 = HathorWallet()` -> w1 has a connection (c1) and a storage (s1)
`w2 = HathorWallet()` -> w2 has a connection (c2) and a storage (s2)

When c1 receives a new history in the ws, it should update s1 and not s2, so it updates the current storage in the `StorageProxy`, so all methods we already have that were not refactored will use the correct storage.

Reply:
@msbrogli I've removed the StorageProxy and kept only the storage as a simpler solution. Can you see what you think?

Reply:
Seems good. Thanks!